vars/runBenchmarkTestScript.groovy

```diff
@@ -0,0 +1,100 @@
+            s3Download(file: "benchmark.ini", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${BENCHMARK_TEST_CONFIG_LOCATION}/${benchmark_config}", force: true)
+
+            /*Added sleep to let the file get downloaded first before write happens. Without the sleep the write is
+            happening in parallel to download resulting in file not found error
+             */
+            sleep(5)
+        }
+    }
+    editBenchmarkConfig("${WORKSPACE}/benchmark.ini")
+    String userTags = getMetadataTags(args.userTag, buildManifest)
+
+    sh([
+            './test.sh',
+            'benchmark-test',
+            "--bundle-manifest ${args.bundleManifest}",
+            "--config ${WORKSPACE}/config.yml",
+            "--workload ${args.workload}",
+            "--benchmark-config ${WORKSPACE}/benchmark.ini",
+            "--user-tag ${userTags}",
+            args.insecure.toBoolean() ? "--without-security" : "",
+            args.singleNode.toBoolean() ? "--single-node" : "",
+            args.minDistribution.toBoolean() ? "--min-distribution" : "",
+            args.useFiftyPercentHeap.toBoolean() ? "--use-50-percent-heap" : "",
+            args.captureNodeStat.toBoolean() ? "--capture-node-stat" : "",
+            isNullOrEmpty(args.suffix) ? "" : "--suffix ${args.suffix}",
+            isNullOrEmpty(args.managerCount) ? "" : "--manager-node-count ${args.managerCount}",
+            isNullOrEmpty(args.dataCount) ? "" : "--data-node-count ${args.dataCount}",
+            isNullOrEmpty(args.clientCount) ? "" : "--client-node-count ${args.clientCount}",
+            isNullOrEmpty(args.ingestCount) ? "" : "--ingest-node-count ${args.ingestCount}",
+            isNullOrEmpty(args.mlCount) ? "" : "--ml-node-count ${args.mlCount}",
+            isNullOrEmpty(args.workloadParams) ? "" : "--workload-params ${args.workloadParams}",
+            isNullOrEmpty(args.additionalConfig) ? "" : "--additional-config ${args.additionalConfig}",
+            isNullOrEmpty(args.dataStorage) ? "" : "--data-node-storage ${args.dataStorage}",
+            isNullOrEmpty(args.mlStorage) ? "" : "--ml-node-storage ${args.mlStorage}",
+            isNullOrEmpty(args.jvmSysProps) ? "" : "--jvm-sys-props ${args.jvmSysProps}"
+    ].join(' '))
+
+}
+
+void editBenchmarkConfig(String config_file) {
+    withCredentials([string(credentialsId: 'beta_datastore_username', variable: 'DATASTORE_USER'),
+                     string(credentialsId: 'beta_datastore_password', variable: 'DATASTORE_PASSWORD')]) {
+        def file = readFile(file: "${config_file}")
+        def contents = file.replace("insert_user_here", "${DATASTORE_USER}")
+        contents = contents.replace("insert_password_here", "${DATASTORE_PASSWORD}")
+        writeFile file: "${config_file}", text: contents
+    }
+
+}
+
+String getMetadataTags(tags, buildManifest) {
+    def metadataTags = "jenkins-build-id:${buildManifest.getArtifactBuildId()},arch:${buildManifest.getArtifactArchitecture()}," +
```

Comment:
`jenkins-build-id` indicates the build number of the jenkins workflow where this will be run. Usually it means that.
How about `distribution-build-id` or `dist-build-id`?

Reply:
makes sense, updated.